include/swift/Syntax/RawSyntax.h

```diff
@@ -120,6 +120,27 @@ using llvm::StringRef;
 #define syntax_assert_token_is(Tok, Kind, Text)
 #endif
 
+namespace {
+/// If the \p Str is not allocated in \p Arena, copy it to \p Arena and adjust
+/// \p Str to point to the string's copy in \p Arena.
+void copyToArenaIfNecessary(
+    StringRef &Str, const swift::RC<swift::syntax::SyntaxArena> &Arena) {
```

Comment:
Could this be a `SyntaxArena` method? I'm just uncomfortable with having an anonymous namespace in a header.

Reply:
Good idea. Then we could also use this to copy the entire source code over here: https://github.com/apple/swift/blob/main/lib/SyntaxParse/SyntaxTreeCreator.cpp#L42